src/registrar/views/domains_json.py

```diff
@@ -0,0 +1,57 @@
+from django.http import JsonResponse
+from django.core.paginator import Paginator
+from registrar.models import UserDomainRole, Domain
+
+
+def get_domains_json(request):
```

Comment:
**blocking question** why aren't we using login required here and in the domain_request one?

Reply:
Lines 10 and 11? Or do you mean that this method should be a class inheriting a perms class?

Reply:
I meant the login_required library which handles the redirect to 403 automatically if you aren't logged in.

Right now on src/api/views the available get function has @login_not_required and for this case you should just have to do @login_required as opposed to handling login requirements in a different way than we do elsewhere in the code.